### PR TITLE
Write metadata and use cached files without youtube-dl, fix youtube-dl processing

### PR DESCRIPTION
**Pull request by Lartza**
Fixes youtube-dl processing. Currently botamusique instantly sets the filename for youtube-dl to mp3 so it downloads whatever it gets (ex. WebM) and calls that an mp3, while the actual post process conversion fails.

Implements writing metadata to files using a youtube-dl postprocessor.
Checks if a file exists first before downloading it, then reads id3 tags from it and returns them without ever calling youtube-dl. This allows the tmp folder cache to work well.

### CODE REVIEW

**Comment by Lartza** [comment]
Oh and id3 handling is through mutagen which is introduced as a new dependency. It's a ~1MB library but has no additional dependencies.

File: mumbleBot.py [hunk 1]
```diff
@@ -320,30 +321,35 @@ def launch_next(self):
...
-                try:
-                    info_dict = ydl.extract_info(url)
-                    video_title = info_dict['title']
-                except youtube_dl.utils.DownloadError:
-                    pass
-                else:
-                    break
-        return path, video_title
+        path = var.config.get('bot', 'tmp_folder') + url_hash + ".%(ext)s"
+        mp3 = path.replace(".%(ext)s", ".mp3")
```
Comment (azlux) [inline mumbleBot.py]: Why did you split this line in two ?

Comment (Lartza) [reply mumbleBot.py]: I'm not sure what you mean? You need to give .%(ext)s to youtube-dl but need to check for .mp3 and return it as the path back from the function.

Comment (Lartza) [reply mumbleBot.py]: And to add to this, doing this instead of str.replace(".mp3", ".%(ext)s") seemed nicer

Comment (azlux) [reply mumbleBot.py]: I didn't know that youtube-dl need %(ext)s value.

**Pull request merged**